### PR TITLE
Add js user tests

### PR DESCRIPTION
**Pull request by sandersn**
Add user tests for the javascript packages that illustrate patterns that we don't support yet.

### CODE REVIEW

File: tests/baselines/reference/user/github.log [hunk 1]
```diff
@@ -0,0 +1,8 @@
+Exit Code: 1
+Standard output:
+index.ts(1,25): error TS7016: Could not find a declaration file for module 'github'. '/home/nathansa/ts/tests/cases/user/github/node_modules/github/index.js' implicitly has an 'any' type.
```
Comment (mhegazy) [inline tests/baselines/reference/user/github.log]: so what is the test testing then?

File: tests/baselines/reference/user/npmlog.log [hunk 2]
```diff
@@ -0,0 +1,106 @@
+Exit Code: 1
+Standard output:
+../../../../node_modules/@types/gulp-help/index.d.ts(70,15): error TS2430: Interface 'GulpHelp' incorrectly extends interface 'Orchestrator'.
```
Comment (weswigham) [inline tests/baselines/reference/user/npmlog.log]: This is in _every_ test, but isn't actually in the tests' respective `node_modules` folder. can we fix this?

Comment (weswigham) [reply tests/baselines/reference/user/npmlog.log]: Yeah, I think these tests are missing the `typeRoots` override our other tests have.

Comment (sandersn) [reply tests/baselines/reference/user/npmlog.log]: Ah, that's right. I'll add that.

File: tests/baselines/reference/user/github.log (hunk 1 above)
Comment (sandersn) [reply tests/baselines/reference/user/github.log]: That's not supposed to be part of the test -- github moved to `@octokit/rest`. I'll go ahead and move it as part of this PR.

**Comment by sandersn** [comment]
Note that the fix to typeroots means that the new user tests can no longer see the global identifiers `test`, `suite`, etc that come from Typescript's use of mocha (or is it jasmine?). This is expected.

File: tests/baselines/reference/user/npmlog.log [hunk 3]
```diff
@@ -0,0 +1,103 @@
+Exit Code: 1
+Standard output:
+node_modules/npmlog/log.js(5,11): error TS2539: Cannot assign to '"/home/nathansa/ts/tests/cases/user/npmlog/node_modules/npmlog/log"' because it is not a variable.
```
Comment (weswigham) [inline tests/baselines/reference/user/npmlog.log]: This error right here's a problem. It's not going to be the same anywhere else and the test is always going to fail. @sandersn should we be printing something else here?

**Pull request merged**